### PR TITLE
migrations: add test for importing pull requests in gitea uploader

### PR DESCRIPTION
**Pull request by singuliere**
See original [merge request from Loïc Dachary @ forgefriends](https://lab.forgefriends.org/forgefriends/forgefriends/-/merge_requests/44)

---

The `gitea_uploader.go` is modified in a minimal way to move around 100 lines to the `updateGitForPullRequest`, the code logic is otherwise not modified. Tests are written to provide 100% coverage, except for the system errors from `os.Create` etc.

### CODE REVIEW

File: services/migrations/gitea_uploader_test.go [hunk 1]
```diff
@@ -217,3 +223,301 @@ func TestGiteaUploadRemapExternalUser(t *testing.T) {
...
+		Email: "test@example.com",
+		Name:  "test",
+		When:  time.Now(),
+	}
+	assert.NoError(t, git.CommitChanges(fromRepo.RepoPath(), git.CommitChangesOptions{
+		Committer: &signature,
+		Author:    &signature,
+		Message:   "Initial Commit",
+	}))
+	fromGitRepo, err := git.OpenRepositoryCtx(git.DefaultContext, fromRepo.RepoPath())
```
Comment (lunny) [inline services/migrations/gitea_uploader_test.go:250]: fromGitRepo should be `Close`

Comment (singuliere) [reply services/migrations/gitea_uploader_test.go:250]: Done.

**Review by lunny — APPROVED** [review]

File: modules/log/file.go [hunk 2]
```diff
@@ -243,6 +243,11 @@ func (log *FileLogger) deleteOldLog() {
 	})
 }
 
+// Content returns the content accumulated in the content provider
+func (log *FileLogger) Content() (string, error) {
+	return log.Filename, nil
```
Comment (Gusted) [inline modules/log/file.go]: In this function it seems like you rather need to return the contents of the file?

File: services/migrations/gitea_uploader_test.go [hunk 3]
```diff
@@ -217,3 +223,303 @@ func TestGiteaUploadRemapExternalUser(t *testing.T) {
...
+	defer fromGitRepo.Close()
```
Comment (Gusted) [inline services/migrations/gitea_uploader_test.go]: You first might want to assert that there is no error, as when a error occur `fromGitRepo` would be nil.

File: services/migrations/gitea_uploader_test.go [hunk 4]
```diff
@@ -217,3 +223,303 @@ func TestGiteaUploadRemapExternalUser(t *testing.T) {
...
+	assert.NoError(t, err)
+	assert.NoError(t, os.WriteFile(filepath.Join(forkRepo.RepoPath(), "README.md"), []byte(fmt.Sprintf("# branch2 %s", forkRepo.RepoPath())), 0o644))
+	assert.NoError(t, git.AddChanges(forkRepo.RepoPath(), true))
+	assert.NoError(t, git.CommitChanges(forkRepo.RepoPath(), git.CommitChangesOptions{
+		Committer: &signature,
+		Author:    &signature,
+		Message:   "branch2 commit",
+	}))
+	forkGitRepo, err := git.OpenRepositoryCtx(git.DefaultContext, forkRepo.RepoPath())
+	defer forkGitRepo.Close()
```
Comment (Gusted) [inline services/migrations/gitea_uploader_test.go]: Ditto here.

File: services/migrations/gitea_uploader_test.go (hunk 3 above)
Comment (singuliere) [reply services/migrations/gitea_uploader_test.go]: Good catch, fixed.

File: services/migrations/gitea_uploader_test.go (hunk 4 above)
Comment (singuliere) [reply services/migrations/gitea_uploader_test.go]: Fixed.

File: modules/log/file.go (hunk 2 above)
Comment (singuliere) [reply modules/log/file.go]: You are correct. Fixed.

**Review by Gusted — APPROVED** [review]

**Comment by singuliere** [comment]
make l-g-t-m work

**Comment by singuliere** [comment]
make l-g-t-m work

**Pull request merged**